src/components/admin/form-components/withFalcorSearch.jsx

```diff
@@ -7,33 +7,30 @@ import { withFalcor } from 'components/hocs/falcor-hocs';
 import { debounce } from 'lib/utilities';
 import { getDisplayName } from 'lib/higher-order-helpers';
 
-export const withSearchableConcept = (
-  fields,
-  formatter,
-  category,
-) => WrappedField => {
+export const withFalcorSearch = (fields, formatter, mode) => WrappedField => {
   const debounceTime = 250;
   class Searchable extends React.Component {
     constructor(props) {
       super(props);
-      this.state = {
-        suggestions: [],
-      };
+      this.getSuggestions = this.getSuggestions.bind(this);
+      this.debounceSuggestions = this.debounceSuggestions.bind(this)();
+    }
 
-      this.handleChange = this.handleChange.bind(this);
-      this.handleClick = this.handleClick.bind(this);
+    getSuggestions(query) {
+      return Promise.resolve(this.debounceSuggestions(query));
```

Comment:
I'm pretty confident that what this does is return a Promise that resolves to a promise that you'll then have to check the resolved value for. There is no need for this wrapper, debounceSuggestions is already returning a promise right?